conf/nginx-ssl.conf

```diff
@@ -1,7 +1,8 @@
 # We track the Mozilla "intermediate" compatibility TLS recommendations.
 # Note that these settings are repeated in the SMTP and IMAP configuration.
 # ssl_protocols has moved to nginx.conf in bionic, check there for enabled protocols.
-ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
+ssl_ciphers TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
+
 ssl_dhparam STORAGE_ROOT/ssl/dh2048.pem;
```

Comment:
Apparently, nowadays the recommendation is to use predefined dh groups. See e.g. a discussion [here](https://crypto.stackexchange.com/questions/81992/should-i-use-self-generated-or-predefined-rfc-7919-dh-groups) or look at the [rfc](https://www.rfc-editor.org/rfc/rfc7919). The idea behind this is that a randomly generated group *may* be less secure than a known predefined one.
If we agree with that reasoning a predefined dh group can be obtained here: https://ssl-config.mozilla.org/ffdhe4096.txt

Reply:
You are reading my mind on this one--as author of the patch I considered it and read the same things, but didn't make the change because there did not seem to be consensus on the necessity of doing it. That said, I think another benefit of doing this is upgrading to the 4096 bit curves and it passes the security sniff test of 'no magic numbers' in cryptography.  I've updated the appropriate configs to generate a ffdhe4096.pem file and the associated configurations to point to it. I've also deleted the smaller key size from the server.

Reply:
I agree that 4096 DH params are a good idea – I've been using them for years – however, generating them takes a very long time (like 30 mins+), so it's worth showing a warning, or something to indicate this to the user.

Reply:
In this case, we don't have to wait because we are using the RFC values.